### PR TITLE
CI(dispatch): Switch to merge commit sha where possible

### PR DESCRIPTION
**Pull request by mattgauntseo-sentry**
Test run: https://github.com/getsentry/sentry/actions/runs/4010879759/jobs/6887862109

### CODE REVIEW

File: .github/workflows/scripts/wait-for-merge-commit.js
Comment (asottile-sentry) [inline .github/workflows/scripts/wait-for-merge-commit.js:26]: if mergeable is ever false you can stop immediately since it's a merge conflict

Comment (mattgauntseo-sentry) [reply .github/workflows/scripts/wait-for-merge-commit.js]: Logic has changed to wait until mergable is non-null and then exit the loop.

I've changed the sentry sha to fallback to the head sha if there is no merge commit, which seems to happen on this PR, maybe because the PR cannot be merged until it's approved?

The logs for both the wait for merge script and the getsentry dispatch should help us quickly debug this in the future too if issues arise.

Comment (mattgauntseo-sentry) [reply .github/workflows/scripts/wait-for-merge-commit.js:28]: OK fixed my broken logic that meant I wasn't getting the commit.

I've cause this script to fail if the PR isn't mergable so the mergeCommitSha should be a reliable output.

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**
